### PR TITLE
Feature/stackn 303

### PR DESCRIPTION
**Pull request by Wrede**
Fixtures are now loaded and django superuser created by a post-install chart hook. The corresponding values are "fixtures" and "studio.django".

Values.yaml has been updated which was obsolete.

Has only been tested with local microk8s. 

Please test on cluster setups. 

UPDATE
**requested changes  and TODO**
- [x] Use already created secret and support for existing secret for subchart postgresql
- [x] Enable helper template to fetch values from subcharts. Requires helm v2.7.0
- [x] remove support for postgresql-ha (prepare for it in future release)
***Create new secret and support for existing secret which contain:***
- [x] 1. Django admin password
- [x] 2. RabbitMQ password
- [X] Overview options for keycloak secret, postgresql for keycloak is it's own subchart within keycloak (bitnami but old version). Possibly use same secret as for studio postgresql to avoid having two postgresql secrets
- [X] Overview secret and/or auto generated password for oidc client secret
- [x] Delete old studio db yaml files
- [x] Cleanup values.yaml

For oidc client secret, since the secret is store isn realm.json as a secret as well as defined i studio configmap, it was hard to create a helper template which can generate a random secret. I don't know why the whole realm.json is stored in a secret. Feel like most of it should be in a configmap and only the sensitive iformation in a secret. Thus, the only weak spot of the current values.yaml is the clientSecret. I added a comment to override this in production.

### CODE REVIEW

**Review by carmat88 — DISMISSED** [review]
Looks good to me. Tested locally and it works well. In a single node setup in a cluster should also work fine.

Very nice with the details about requirement and notes at the beginning of the values.yaml file.

File: scaleout/stackn/values.yaml [hunk 1]
```diff
@@ -1,6 +1,96 @@
...
+
+#NOTES
+# - For local development/testing consider setting "oidc.verify_ssl" to false 
+
+#node anchors used to reduce manual configuration
+global:
+  db:
+    name: &db_name stackn
+    user: &db_user stackn
+    password: &db_password fdfweorkpokf931d0vsdf9
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:20]: Password in this line?

File: scaleout/stackn/values.yaml [hunk 2]
```diff
@@ -43,115 +130,289 @@ studio:
...
-      cpu: 400m
-      memory: 2Gi
-    requests:
-      cpu: 200m
-      memory: 1Gi
 
 # default credentials for rabbitmq. override in production!
 rabbit:
   username: admin
   password: LJqEG9RE4FdZbVWoJzZIOQEI
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:195]: Remove this and use a placeholder.

File: scaleout/stackn/values.yaml [hunk 3]
```diff
@@ -225,78 +483,103 @@ keycloak:
...
+
   extraEnv: |
     - name: KEYCLOAK_IMPORT
       value: /realm/realm.json
     - name: KEYCLOAK_USER
-      value: admin
+      value: keycloak
     - name: KEYCLOAK_PASSWORD
-      value: password
+      value: MxaspoewFEWERLLGBasde
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:493]: Password in this line?

File: scaleout/stackn/values.yaml [hunk 4]
```diff
@@ -225,78 +483,103 @@ keycloak:
...
+          - pods
+        verbs:
+          - get
+          - list
+
   postgresql:
+    enabled: true
     postgresqlUsername: keycloak
-    postgresqlPassword: db_password
+    postgresqlPassword: kd93l.s/2094lfOSd;d
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:511]: Password in this line?

File: scaleout/stackn/values.yaml [hunk 5]
```diff
@@ -225,78 +483,103 @@ keycloak:
...
+
 prometheus:
-  enabled: true
+  enabled: false
   server:
     ingress:
       enabled: true
       hosts:
-        - prometheus.your-domain.com
+        - prometheus.10.0.144.239.nip.io
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:538]: ip here

File: scaleout/stackn/values.yaml [hunk 6]
```diff
@@ -225,78 +483,103 @@ keycloak:
...
       tls:
         - secretName: prod-ingress
           hosts:
-            - prometheus.your-domain.com
+            - prometheus.10.0.144.239.nip.io
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:542]: ip here

File: scaleout/stackn/values.yaml [hunk 7]
```diff
@@ -225,78 +483,103 @@ keycloak:
...
+      - ReadWriteMany
+  
   args:
     - "-Dkeycloak.profile.feature.token_exchange=enabled"
 
+oidc:
+    enabled: true
+    realm: STACKn
+    client_id: studio
+    client_secret: Q#cdsIA(2MDAKLS00caradsf<OLAL:KSDn(kvd
```
Comment (morganekmefjord) [inline scaleout/stackn/values.yaml:527]: set placehodler

**Review by morganekmefjord — CHANGES_REQUESTED** [review]

**Comment by morganekmefjord** [comment]
An improvement here would be to use
```bash
{{ randAlphaNum 20 | b64enc }}
```
to generate passwords on deploy

File: scaleout/stackn/values.yaml (hunk 1 above)
Comment (carmat88) [reply scaleout/stackn/values.yaml:20]: Thank you Morgan for noticing these lines!

**Comment by Wrede** [comment]
> An improvement here would be to use
> 
> ```shell
> {{ randAlphaNum 20 | b64enc }}
> ```
> 
> to generate passwords on deploy

@morganekmefjord  Ok but won't this be a problem during helm upgrade? Check [here](https://helm.sh/docs/howto/charts_tips_and_tricks/#be-careful-with-generating-random-values)

**Comment by Wrede** [comment]
Looking into secrets...

File: scaleout/stackn/requirements.yaml [hunk 8]
```diff
@@ -42,10 +42,9 @@ dependencies:
   - name: postgresql-ha
     version: 7.3.0
     repository: https://charts.bitnami.com/bitnami
-    condition: postgresql-ha.enabled
+    condition: postgres-ha.enabled
```
Comment (morganekmefjord) [inline scaleout/stackn/requirements.yaml:45]: Inconsistent naming?

File: scaleout/stackn/templates/celery-beat-deployment.yaml [hunk 9]
```diff
@@ -29,7 +29,7 @@ spec:
       initContainers:
         - name: wait-for-db
           image: postgres
-          command: ['sh', '-c', 'until pg_isready --host={{ .Values.postgres.host }}.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local --port={{ .Values.postgres.port }}; do echo waiting for database; sleep 2; done;']
+          command: ['sh', '-c', 'until pg_isready --host={{ include "stackn.postgres.host" . }}.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local --port={{ include "stackn.postgres.port" . }}; do echo waiting for database; sleep 2; done;']
```
Comment (morganekmefjord) [inline scaleout/stackn/templates/celery-beat-deployment.yaml:32]: Is "." at end required?

**Comment by Wrede** [comment]
The current values.yaml only requires the user to change domain and cluster config.

Note:
For oidc client secret, since the secret is store isn realm.json as a secret as well as defined i studio configmap, it was hard to create a helper template which can generate a random secret. I don't know why the whole realm.json is stored in a secret. Feel like most of it should be in a configmap and only the sensitive iformation in a secret. Thus, the only weak spot of the current values.yaml is the clientSecret. I added a comment to override this in production.

File: scaleout/stackn/requirements.yaml (hunk 8 above)
Comment (Wrede) [reply scaleout/stackn/requirements.yaml:45]: thanks for finding this typo!

Comment (Wrede) [reply scaleout/stackn/requirements.yaml]: fixed in [commit](https://github.com/scaleoutsystems/charts/pull/72/commits/f6aeb0a37d4ebf63e1b01f9c2d018f71e6318745)

File: scaleout/stackn/templates/celery-beat-deployment.yaml (hunk 9 above)
Comment (Wrede) [reply scaleout/stackn/templates/celery-beat-deployment.yaml:32]: Yes "include" requires two arguments. I don't know exactly how go templates work but the "." is special type of variable of any type, i.e it returns the complete object from "stackn.postgres.host".

**Comment by carmat88** [comment]
> Note: For oidc client secret, since the secret is store isn realm.json as a secret as well as defined i studio configmap, it was hard to create a helper template which can generate a random secret. I don't know why the whole realm.json is stored in a secret. Feel like most of it should be in a configmap and only the sensitive iformation in a secret. Thus, the only weak spot of the current values.yaml is the clientSecret. I added a comment to override this in production.

Indeed, this is something we can maybe look into for the next release. It is already great that password and secrets are not showing up directly hard-coded in the `value.yaml` file (except for oidc as explained above). Very important to have comments on how to override this in production, so that's great.

**Review by carmat88 — APPROVED** [review]

**Pull request merged**